### PR TITLE
add --greetingbotcommand

### PR DESCRIPTION
**Pull request by wonderingabout**
due to its nature, this supports all bots

tested to work, ready to merge
https://beta.online-go.com/game/7599

greetingbotcommand 2

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -36,8 +36,9 @@ exports.updateFromArgv = function() {
         .demand('apikey')
         .describe('username', 'Specify the username of the bot, for example GnuGo')
         .describe('apikey', 'Specify the API key for the bot')
-        .describe('greeting')
+        .describe('greeting', 'Greeting message to appear in chat at first move (ex: -Hello, have a nice game-)')
         .string('greeting')
+        .describe('greetingbotcommand', `Additionnal greeting message displaying bot command, in OGS games`)
```
Comment (sanderland) [inline config.js:41]: missspellllleddd

File: game.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ class Game {
         this.state = null;
         this.opponent_evenodd = null;
         this.greeted = false;
+        this.greetedbotcommand = false;
```
Comment (sanderland) [inline game.js:20]: i think these can be merged into one

Comment (wonderingabout) [reply game.js:20]: as long as we do our greeting only once yes
but if someday we want to move this botcommand greeting at another place, then we'd need a specific option

for now your advice seems the most fit to this case :100:

Comment (wonderingabout) [reply game.js:20]: edit: it makes the code a bit less clear, should we remove this, what does everyone think?

Comment (Dorus) [reply game.js:20]: Yes, merge this into one, and move `!this.greeted` in the check on line 371 to the check on line 370.

Comment (wonderingabout) [reply game.js:20]: tested to work
but do not merge yet, i need to commit last changes

Comment (wonderingabout) [reply game.js]: rebased and tested, if travis is good we can merge this

File: game.js [hunk 3]
```diff
@@ -366,9 +366,15 @@ class Game {
             return;
         if (this.state.phase !== 'play')
             return;
-        if( config.greeting && !this.greeted && this.state.moves.length < (2 + this.state.handicap) ){
-            this.sendChat( config.greeting, "discussion");
-            this.greeted = true;
+        if (!this.greeted && this.state.moves.length < (2 + this.state.handicap)) {
+            if (config.greeting) {
+                this.sendChat(config.greeting, "discussion");
+                this.greeted = true;
```
Comment (sanderland) [inline game.js:372]: outside of the if

Comment (wonderingabout) [reply game.js]: thanks, fixed in https://github.com/online-go/gtp2ogs/pull/274/commits/d57c8797786ea6da968473e0fe18cb71b2cf66e8

File: config.js [hunk 4]
```diff
@@ -36,8 +36,9 @@ exports.updateFromArgv = function() {
         .demand('apikey')
         .describe('username', 'Specify the username of the bot, for example GnuGo')
         .describe('apikey', 'Specify the API key for the bot')
-        .describe('greeting')
+        .describe('greeting', 'Greeting message to appear in chat at first move (ex: -Hello, have a nice game-)')
```
Comment (Dorus) [inline config.js:39]: Shouldn't this be `--greeting -Hello, have a nice game-`. I find it confusing to leave out the argument name, but start the example with a dash anyway.

Comment (wonderingabout) [reply config.js:39]: i dont think so, in https://github.com/wonderingabout/gtp2ogs/blob/greeting-botcommand/docs/OPTIONS-LIST.md#greeting we provide `"Hello, have a nice game"` as an example of greeting message

we are not trying to give an example of how to use the gtp2ogs argument, just an example of messages bot admin could use

it is also the same for farewell https://github.com/online-go/gtp2ogs/blob/devel/docs/OPTIONS-LIST.md#farewell

Comment (wonderingabout) [reply config.js:39]: we could remove the examples but i think new bot admins would find them helpful

Comment (Dorus) [reply config.js:39]: yes i'm okay with a sample, but please use `Hello, have a nice game` instead of `-Hello, have a nice game-`.

**Comment by Dorus** [comment]
> 
> 
> LGTM

Idem

**Pull request merged**